src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/CommonRoutes.scala

```diff
@@ -38,34 +38,43 @@ private[routes] abstract class CommonRoutes(
     viewCache: ViewCache[Task])(implicit project: Project, indexers: Clients[Task], config: AppConfig) {
 
   import indexers._
-  implicit val acl     = acls
-  implicit val c       = caller
-  implicit val subject = caller.subject
+
+  protected implicit val acl: AccessControlLists                = acls
+  protected implicit val c: Caller                              = caller
+  protected implicit val subject: Identity.Subject              = caller.subject
+  protected implicit def additional: AdditionalValidation[Task] = AdditionalValidation.pass
 
   private[routes] val resourceName = prefix.capitalize
 
-  implicit def additional: AdditionalValidation[Task] = AdditionalValidation.pass
+  private[routes] val simultaneousParamsRejection: AkkaRejection =
+    validationRejection("'rev' and 'tag' query parameters cannot be present simultaneously.")
+
+  protected val readPermission: Set[Permission] = Set(Permission.unsafe("resources/read"))
+
+  protected val writePermission: Set[Permission] =
+    if (prefix == "views") Set(Permission.unsafe("views/write"))
+    else Set(Permission.unsafe("resources/write"))
```

Comment:
why should we have only `views/write` and `resources/write` ?
schema, resolvers and the others might still be interesting